Gemfile

```diff
@@ -1,12 +1,18 @@
 source 'https://rubygems.org'
 
-gem 'solidus', github: 'solidusio/solidus', branch: ENV.fetch('SOLIDUS_BRANCH', 'master')
+branch = ENV.fetch('SOLIDUS_BRANCH', 'master')
+gem 'solidus', github: 'solidusio/solidus', branch: branch
+
+if branch == 'master' || branch >= "v2.0"
```

Comment:
A check against `Rails::VERSION::MAJOR >= 5` here is better, since this has nothing to do with solidus, it's a raisl 5 change.

Additionally we might want to look at avoiding writing any new controller tests and use request specs instead. Not sure if company has committed to that approach, but that's probably a discussion that needs to happen.